File: apps/api/v2/src/main/java/com/crashx/api/repositories/ProjectRepository.java
package com.crashx.api.repositories;

import com.crashx.api.models.Project;
import org.springframework.data.jpa.repository.JpaRepository;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface ProjectRepository extends JpaRepository<Project, UUID> {
    Optional<Project> findByApiKey(String apiKey);
    List<Project> findByUserId(UUID userId);
}

File: apps/api/v1/src/repositories/project_repository.py
from typing import Optional, List
from sqlalchemy import select
from sqlalchemy.ext.asyncio import AsyncSession
from src.models.project import Project
from src.repositories.base_repository import BaseRepository
from sqlalchemy.dialects.postgresql import UUID

class ProjectRepository(BaseRepository[Project]):
    def __init__(self, db: AsyncSession):
        super().__init__(Project, db)

    async def get_by_api_key(self, api_key: str) -> Optional[Project]:
        query = select(Project).where(Project.api_key == api_key)
        result = await self.db.execute(query)
        return result.scalars().first()

    async def get_by_user(self, user_id: UUID) -> List[Project]:
        query = select(Project).where(Project.user_id == user_id)
        result = await self.db.execute(query)
        return result.scalars().all()
